src/update-process-env.js

```diff
@@ -0,0 +1,49 @@
+/** @babel */
+
+import {spawnSync} from 'child_process'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set(['NODE_ENV', 'NODE_PATH'])
+
+export default function updateProcessEnv (launchEnv) {
+  let envToAssign
+  if (launchEnv.PWD) {
+    envToAssign = launchEnv
+  } else {
+    if (process.platform === 'darwin') {
+      envToAssign = getEnvFromShell()
+    }
+  }
+
+  if (envToAssign) {
+    for (let key in process.env) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        delete process.env[key]
+      }
+    }
+
+    for (let key in envToAssign) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        process.env[key] = envToAssign[key]
+      }
+    }
+  }
+}
+
+function getEnvFromShell () {
```

Comment:
Tell me if I'm wrong but I personally believe that a project should be focused and Atom's mission is to be a Text Editor not an env patcher. The reason I suggest this time and time again is that the current method is not reliable and we still have to patch `PATH` in all of the linter providers, if we can get it done in the core it'll make all those linter providers have one less dependency.

While you can copy a little to **save** a dependency, but if you keep doing this you'll end up with a bunch of bloatware that will be hard to maintain. and we can already see that Atom's maintaining power is being wasted into something that's already available as a module.

https://github.com/atom/atom/pull/12562 https://github.com/atom/atom/pull/12396 https://github.com/atom/atom/pull/11436 https://github.com/atom/atom/pull/12378 https://github.com/atom/atom/pull/12317 https://github.com/atom/atom/pull/12028 https://github.com/atom/atom/pull/11054 and more